### PR TITLE
Refactor DOB field and add validation for householdmember age

### LINKED ISSUE #449: Household Member DOB should not have a minimum age
While the primary applicant must be 18 years old to apply, household members do not have a minimum age. 

image

### PR DESCRIPTION
**Pull request by software-project**
Resolves #449

### CODE REVIEW

**Comment by bk3c** [comment]
@software-project Can you make a storybook story for the new DOB component, so we get the documentation and storyshot testing benefits?

**Comment by jaredcwhite** [comment]
@software-project I like this approach. Only thought is the hardcoding of what the keys on the error object are…not sure if we should expect that to be true in all cases (especially if we end up with multiple DOB fields on a single form)

**Comment by software-project** [comment]
@jaredcwhite by error key? You mean translation? I'm passing whole error object, so this should work for two DOBFields per page. @bk3c I was thinking about the storybook, but I wasn't sure how to pass register or errors. Will try to figure that one out.

**Comment by jaredcwhite** [comment]
@software-project nevermind, I didn't realize you were passing a scoped subset of the error object at times. seems fine.

File: shared/ui-components/src/forms/__snapshots__/DOBField.stories.storyshot [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+          >
+            <input
+              className="input"
+              defaultValue=""
+              id="birthMonth"
+              maxLength={2}
+              name="birthMonth"
+              placeholder="MM"
+              type="text"
+            />
```
[the remark this answers is not in the document]
Comment (bk3c) [reply shared/ui-components/src/forms/__snapshots__/DOBField.stories.storyshot:66]: This seems accurate to me. @slowbot, do you have an opinion of what the field label should say here? "Date of Birth"?

@software-project note that we'll need to make the label a translatable string too...

[the remark this answers is not in the document]
Comment (software-project) [reply shared/ui-components/src/forms/__snapshots__/DOBField.stories.storyshot:66]: @bk3c there is a label on top of the component, but we dont have labels for each one of them.

**Comment by jaredcwhite** [comment]
I think we're just waiting for @slowbot's thoughts regarding the accessibility/labels issue?

**Comment by software-project** [comment]
@slowbot how does it look now?

**Pull request merged**